pkg/tsdb/stackdriver/stackdriver.go

```diff
@@ -53,9 +55,25 @@ func NewStackdriverExecutor(dsInfo *models.DataSource) (tsdb.TsdbQueryEndpoint,
 		return nil, err
 	}
 
+	var (
+		timeout         = dsInfo.JsonData.GetPath("circuit", "timeout_ms").MustInt(3000)
+		minReqs         = dsInfo.JsonData.GetPath("circuit", "min_requests").MustInt(3)
+		minFailureRatio = dsInfo.JsonData.GetPath("circuit", "min_failure_ratio").MustFloat64(0.6)
+	)
+
+	breaker := gobreaker.NewCircuitBreaker(gobreaker.Settings{
```

Comment:
`NewStackdriverExecutor` is called for each request so the circuit breaker wouldn't be that useful this way. I would create circuit breaker inside `dsInfo.GetHttpClient()` and implement a RoundTripper wrapping a request since that is cached and/or expose a new `dsInfo.GetCircuitBreaker()` that is cached.

I would consider adding global setting to be able to enable circuit breakers, disabled by default, together with settings and defaults for timeout_ms, min_requests and min_failure_ratio. Settings can possibly be overridden by datasource, but can wait until later. My suggested setup would be a good start for Hosted Grafana.